Tests/AirbnbSwiftFormatToolTests/AirbnbSwiftFormatToolTests.swift

```diff
@@ -0,0 +1,255 @@
+          return EXIT_SUCCESS
+        },
+        swiftLint: {
+          ranSwiftLint = true
+          return SwiftLintExitCode.lintFailure
+        },
+        swiftLintAutocorrect: {
+          ranSwiftLintAutocorrect = true
+
+          // When autocorrecting SwiftLint returns EXIT_SUCCESS
+          // even if there were violations that were fixed
+          return EXIT_SUCCESS
+        }))
+
+    XCTAssertEqual(error as? ExitCode, ExitCode(SwiftFormatExitCode.lintFailure))
+    XCTAssertTrue(ranSwiftFormat)
+    XCTAssertTrue(ranSwiftLint)
+    XCTAssertTrue(ranSwiftLintAutocorrect)
+  }
+
+  func testFormatWithOnlySwiftLintAutocorrectedViolation() {
+    var ranSwiftFormat = false
+    var ranSwiftLint = false
+    var ranSwiftLintAutocorrect = false
+
+    let error = runFormatTool(
+      with: MockCommands(
+        swiftFormat: {
+          ranSwiftFormat = true
+          return EXIT_SUCCESS
+        },
+        swiftLint: {
+          ranSwiftLint = true
+
+          // Assume that the codebase has violations that would be corrected by SwiftLint autocorrect.
+          if ranSwiftLintAutocorrect {
+            // If SwiftLint autocorrect has already run, then there are no more violations.
+            // This is the expected behavior.
+            return EXIT_SUCCESS
+          } else {
+            // If SwiftLint autocorrect hasn't run yet, then there are still violations.
+            // This should not happen, because we run autocorrect first.
+            return SwiftLintExitCode.lintFailure
+          }
+        },
+        swiftLintAutocorrect: {
+          // Assume that this SwiftLint autocorrect invocation applied a code change.
+          // In this case, SwiftLint still returns a zero exit code.
+          ranSwiftLintAutocorrect = true
+          return EXIT_SUCCESS
+        }))
+
+    // Even though there was a SwiftLint failure, it was autocorrected so doesn't require attention.
+    // The tool should not return an error (e.g. it should return a zero exit code).
+    XCTAssertNil(error)
+
+    XCTAssertTrue(ranSwiftFormat)
+    XCTAssertTrue(ranSwiftLint)
+    XCTAssertTrue(ranSwiftLintAutocorrect)
+  }
+
+  func testLintWithViolations() {
+    var ranSwiftFormat = false
+    var ranSwiftLint = false
+    var ranSwiftLintAutocorrect = false
+
+    let error = runFormatTool(
+      arguments: ["--lint"],
+      with: MockCommands(
+        swiftFormat: {
+          ranSwiftFormat = true
+          return SwiftFormatExitCode.lintFailure
+        },
+        swiftLint: {
+          ranSwiftLint = true
+          return SwiftLintExitCode.lintFailure
+        },
+        swiftLintAutocorrect: {
+          ranSwiftLintAutocorrect = true
+          return EXIT_SUCCESS
+        }))
+
+    XCTAssertEqual(error as? ExitCode, ExitCode.failure)
+    XCTAssertTrue(ranSwiftFormat)
+    XCTAssertTrue(ranSwiftLint)
+    XCTAssertFalse(ranSwiftLintAutocorrect)
+  }
+
+  func testLintWithOnlySwiftLintViolation() {
```

Comment:
For completeness, should we have a test case for lint with only SwiftFormat violations?

Reply:
I considered it, but the code for that test case would be identical to the existing `testLintWithNoViolations` test case (since in both cases SwiftFormat just returns `EXIT_SUCCESS`) so I'm not sure it adds much value.

Reply:
@calda I want to make sure we're on the same page. In the case where we run SwiftFormat in lint-only mode it returns a non-zero exit code (see https://github.com/airbnb/swift/pull/242#discussion_r1355831828). I am pretty sure that the new test case would be unique.

```swift
  func testLintWithOnlySwiftFormatViolation() {
    var ranSwiftFormat = false
    var ranSwiftLint = false
    var ranSwiftLintAutocorrect = false

    let error = runFormatTool(
      arguments: ["--lint"],
      with: MockCommands(
        swiftFormat: {
          ranSwiftFormat = true
          return SwiftFormatExitCode.lintFailure
        },
        swiftLint: {
          ranSwiftLint = true
          return EXIT_SUCCESS
        },
        swiftLintAutocorrect: {
          ranSwiftLintAutocorrect = true
          return EXIT_SUCCESS
        }))

    XCTAssertEqual(error as? ExitCode, ExitCode.failure)
    XCTAssertTrue(ranSwiftFormat)
    XCTAssertTrue(ranSwiftLint)
    XCTAssertFalse(ranSwiftLintAutocorrect)
  }
```

Reply:
My mistake, I misinterpreted your comment. It makes sense to add a test for this.